libsmtutil/CHCSmtLib2Interface.h

```diff
@@ -79,17 +78,19 @@ class CHCSmtLib2Interface: public CHCSolverInterface
 	/// Translates CHC solver response with a model to our representation of invariants. Returns None on error.
 	std::optional<smtutil::Expression> invariantsFromSolverResponse(std::string const& response) const;
 
+	/// Hook to setup external solver call
+	virtual void setupSmtCallback() {}
```

Comment:
Should probably `smtAssert(false)` here.

Reply:
We cannot do that yet, because `--model-checker-solvers smtlib2` is a valid option now, and it will instantiate this base class.
What will happen is that it will try to call `SMTCommand`, but that will do nothing.

I think eventually we should just drop `smtlib2` as a possible choice for the solver (once also z3 migrates to the new usage).